python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1849,6 +1849,39 @@ def get_can_run_with_parent_subsets(
 
     is_self_dependency = parent_asset_key == candidate_asset_key
 
+    if not (
+        # this check is here to guard against cases where the parent asset has a superset of
+        # the child asset's asset partitions, which will mean that the runs that would be created
+        # would not combine the parent and child assets into a single run. this is not relevant
+        # for self-dependencies, because the parent and child are the same asset.
+        is_self_dependency
+        or (
+            # in the typical case, we will only allow this candidate subset to be requested if
+            # it contains exactly the same partitions as its parent asset for this evaluation,
+            # otherwise they may end up in different runs
+            parent_being_requested_this_tick_subset.get_internal_value()
+            == entity_subset_to_filter.get_internal_value()
+        )
+        or (
+            # for non-subsettable multi-assets, we will not have yet requested the parent asset
+            # partitions, so we just check that we have a matching set of partitions
+            asset_graph_view.get_entity_subset_from_asset_graph_subset(
+                candidate_asset_graph_subset_unit, parent_asset_key
```

Comment:
Confirming understanding - the only reason any additional keys other than the candidate asset key would be in this candidate_asset_graph_subset_unit would be if the candidate asset is part of a non-subsettable multi-asset?

Reply:
if that's right this is just a very very confusing variable name (not this PR's fault) - i feel like the name should have like contained_non_subsettable_multi_asset in int somewhere

Reply:
>  the only reason any additional keys other than the candidate asset key would be in this candidate_asset_graph_subset_unit would be if the candidate asset is part of a non-subsettable multi-asset

exactly right

> if that's right this is just a very very confusing variable name

also exactly right